walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -25,12 +25,36 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     List<TimeRange> availableTimes = new ArrayList<>();
     
+    if (!request.getOptionalAttendees().isEmpty()) {
```

Comment:
This is incredibly clean, well done. I think the only thing that it doesn't do is explain the "business logic", as your function names and variables don't exactly explain why everything is happening. You COULD try to rename and restructure things, but I think since you've written a small concise solution, I would recommend each conditional just have a comment above it that explains the "business logic" i.e. why are we doing this check, what is the expect outcome sort of thing.

Reply:
Just a comment like:
```java
// First we try to find a meeting time that works for all attendees, including optional.
```